src/OpenTelemetry.AutoInstrumentation/Configurations/FileBasedConfiguration/YamlAttribute.cs

```diff
@@ -0,0 +1,227 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Globalization;
+using OpenTelemetry.AutoInstrumentation.Logging;
+using Vendors.YamlDotNet.Serialization;
+
+namespace OpenTelemetry.AutoInstrumentation.Configurations.FileBasedConfiguration;
+
+internal class YamlAttribute
+{
+    private static readonly IOtelLogger Log = OtelLogging.GetLogger();
+
+    /// <summary>
+    /// Gets or sets the name of the resource attribute.
+    /// </summary>
+    [YamlMember(Alias = "name")]
+    public string? Name { get; set; }
+
+    /// <summary>
+    /// Gets or sets the value of the resource attribute.
+    /// </summary>
+    [YamlMember(Alias = "value")]
+    public object? Value { get; set; }
+
+    /// <summary>
+    /// Gets or sets the type of the resource attribute.
+    /// </summary>
+    [YamlMember(Alias = "type")]
+    public string Type { get; set; } = "string";
+
+    public static List<KeyValuePair<string, object>> ParseAttributes(List<YamlAttribute>? attributes)
```

Comment:
Can we simply return a `TagList` rather than return a `List<KeyValuePair<string, object>>`?

```suggestion
    public static TagList ParseAttributes(List<YamlAttribute>? attributes)
```

Reply:
fixed in 378f52a